paramiko/hostkeys.py

```diff
@@ -220,15 +227,15 @@ def __iter__(self):
     def __len__(self):
         return len(self.keys())
 
-    def __delitem__(self, key):
-        k = self[key]
-
     def __getitem__(self, key):
         ret = self.lookup(key)
         if ret is None:
             raise KeyError(key)
         return ret
 
+    def __delitem__(self, key):
+        pass  # Needed for instantiating HostKeys.
```

Comment:
a minor functional change here - previously an exception would be thrown if the key was not found, now it won't